src/_posts/2020-05-19-promises-and-lies.md

```diff
@@ -0,0 +1,149 @@
+```
+
+It is possible to return a promise from a promise so that they can be chained. Note that the catch at the end of a chain will catch whichever reject happens first.
+
+Once a promise has resolved to a value, then it will always resolve to the same value. You can, therefore, call `then` on a given promise multiple times and it will always return the same result. I'll come back to this concept later.
+
+Promises also have no means of being cancelled. The only output of them is through actions of the resolve or reject functions. This means that it is possible to cache a promise - just note that the cached promise could have already failed!
+
+## The Promise class
+
+The javascript promise class has the following static functions:
+
+```Javascript
+Promise.all( ) 
+```
+
+This will turn iterable of Promises into a Promise that will either return an iterable of the success values or the first failure condition.
+
+```Javascript
+Promise.allSettled()
+```
+
+This will take an iterable of Promises and returns a Promise that will return an iterable of the results (either resolved or rejected).
+
+```Javascript
+Promise.race()
+```
+
+This will take an iterable of Promises and returns a Promise that will return the result of the first resolved or rejected Promise.
+
+```Javascript
+Promise.reject()
+```
+
+This returns a Promise that has been rejected with the given reason.  
+
+```Javascript
+Promise.resolve()
+```
+
+This will either return the result of a supplied promise or will return a Promise that has resolved the supplied data.
+
+## What is missing?
+
+The standard library has no built-in concept of timeout. This means that each user must consider (or more likely ignore) this important concept. Without a controlled timeout you have no means of ensuring that a given `then()` will ever return.
+
+Serverless functions are one example of where a fixed timeout is essential. The serverless framework adds a timeout value that defaults to 6 seconds. It is important that you get to choose how you handle failure rather than just having the function fail.
+More importantly, you are charged on execution time so waiting for a function to return costs money. 
+
+## Introducing lie.js
+
+This is where [lie.js](https://www.npmjs.com/package/lie.js) comes in. It provides a standard means of adding timeouts to Promises. These are very small functions that bring in no additional dependencies.
+
+It currently provides two helper functions that are intended to be used in a Promise.race with an existing promise.
+
+### orReject
+
+```javascript
+Promise.race([yourPromise, orReject('bang', 1)])
+```
+
+This will reject the promise with the given value after the timeout.
+
+### orTimeout
+
+```javascript
+Promise.race([yourPromise, orTimeout('bang', 1)])
+```
+
+This will succeed the promise with the given value after the timeout.
+
+These two simple functions allow you to add timeout functionality to any Javascript promise.
```

Comment:
Capitalization

These two simple functions allow you to add timeout functionality to any Javascript Promise.